build.gradle

```diff
@@ -1,37 +1,63 @@
-//------------------------------------------- PLUGIN DEFINITION -----------------------------------------------//
-
-plugins {
-    id "java-gradle-plugin"
-    id "eclipse"
-    id "jacoco"
+/**
+Template file for classes using GradleRIO
+@author Jaci
+*/
+
+
+buildscript {
+  repositories {
+    //mavenLocal()
+	mavenCentral()
+	maven {
+		name = "GradleRIO"
+		url = "http://dev.imjac.in/maven"
+	}
+  }
+  dependencies {
+    classpath group: 'jaci.openrio.gradle', name: 'GradleRIO', version: '+'			//Change this line if you wish to Update GradleRIO
+//    compile 'org.apache.commons:commons-collections4:4.1'
+//    testCompile group: 'junit', name: 'junit', version: '4.+'
+  }
 }
 
-//------------------------------------------- VERSION MANAGEMENT -----------------------------------------------//
-
-version = '1.0'
-sourceCompatibility = 1.8
-targetCompatibility = 1.8
+//plugins {
+//    id 'java'
+//    id 'java-gradle-plugin'
+//    id 'idea'
+//    id 'eclipse'
+//    id 'jacoco'
+//}
+apply plugin: 'java'
+apply plugin: 'java-gradle-plugin'
+apply plugin: 'idea'
+apply plugin: 'eclipse'
+apply plugin: 'GradleRIO'                                 //Apply the GradleRIO plugin
+apply plugin: 'jacoco'
+
+gradlerio.robotClass = "org.usfirst.frc.team4571.Robot"   //The class for the main Robot Class. Used in manifest
+gradlerio.team = "4571"                                   //Your FRC team number (e.g. 5333 for team 'Can't C#', or 47 for Chief Delphi)
+gradlerio.rioIP = "10.45.71.20"                         //Uncomment to specify the IP address of the RIO
 
-//------------------------------------------- DEPENDENCY MANAGEMENT -----------------------------------------------//
+def robotManifest = {
+  attributes 'Main-Class': 'edu.wpi.first.wpilibj.RobotBase'
+  attributes 'Robot-Class': gradlerio.robotClass
+  attributes 'Implementation-Title': 'Crate Robot'
+}
 
-// Defining the dependency repository locations
-repositories {
-	// If you are behind a firewall and have a local maven repo set up, then use
-    mavenLocal()
-    
-    // If you aren't behind a firewall, then use
-    mavenCentral(); 
+jar {
+  from configurations.compile.collect { it.isDirectory() ? it : zipTree(it) }
+  manifest robotManifest
 }
 
-dependencies {
-	compile files('libs/WPILib.jar', 'libs/NetworkTables.jar')
-	compile 'org.apache.commons:commons-collections4:4.1'
-    testCompile group: 'junit', name: 'junit', version: '4.+'
+task genJavadoc(type: Jar, dependsOn: javadoc) {
```

Comment:
I dont quite understand how we will get the dependencies that we are removing here. If we get rid of the compile files lines, I am expecting the Travis build to break. Isnt that right?